crates/accelerate/src/two_qubit_decompose.rs

```diff
@@ -180,20 +263,792 @@ fn __num_basis_gates(basis_b: f64, basis_fidelity: f64, unitary: MatRef<c64>) ->
     traces
         .into_iter()
         .enumerate()
-        .map(|(idx, trace)| (idx, trace_to_fid(&trace) * basis_fidelity.powi(idx as i32)))
+        .map(|(idx, trace)| {
+            (
+                idx,
+                trace_to_fid_c64(&trace) * basis_fidelity.powi(idx as i32),
+            )
+        })
         .min_by(|(_idx1, fid1), (_idx2, fid2)| fid2.partial_cmp(fid1).unwrap())
         .unwrap()
         .0
 }
 
 /// Average gate fidelity is :math:`Fbar = (d + |Tr (Utarget \\cdot U^dag)|^2) / d(d+1)`
 /// M. Horodecki, P. Horodecki and R. Horodecki, PRA 60, 1888 (1999)
-fn trace_to_fid(trace: &c64) -> f64 {
+fn trace_to_fid_c64(trace: &c64) -> f64 {
     (4.0 + trace.faer_abs2()) / 20.0
 }
 
+fn trace_to_fid(trace: Complex64) -> f64 {
+    (4.0 + trace.abs().powi(2)) / 20.0
+}
+
+/// A good approximation to the best value x to get the minimum
+/// trace distance for :math:`U_d(x, x, x)` from :math:`U_d(a, b, c)`.
+fn closest_partial_swap(a: f64, b: f64, c: f64) -> f64 {
+    let m = (a + b + c) / 3.;
+    let [am, bm, cm] = [a - m, b - m, c - m];
+    let [ab, bc, ca] = [a - b, b - c, c - a];
+    m + am * bm * cm * (6. + ab * ab + bc * bc * ca * ca) / 18.
```

Comment:
as @levbishop pointed out there is a misprint in this formula.
It should be:
` m + am * bm * cm * (6. + ab * ab + bc * bc + ca * ca) / 18.`
See https://github.com/Qiskit/qiskit/pull/11953